### PR TITLE
Align checkout with specs

### PR DESCRIPTION
**Pull request by nabeelamirch**
Aligns the checkout object with the specs

resolves: GCI-747

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/orders/api/model/OrderData.java [hunk 1]
```diff
@@ -8,6 +8,8 @@
 
     private ActionedBy orderedBy;
 
+    private OrderLinks links;
+
     public LocalDateTime getOrderedAt() {
```
Comment (VaughanJackson123) [inline src/main/java/uk/gov/companieshouse/orders/api/model/OrderData.java]: Is `ordered_at` to be removed?

Comment (nabeelamirch) [reply src/main/java/uk/gov/companieshouse/orders/api/model/OrderData.java]: Although it is removed from the Checkout object, it still exists in the Order object.

File: src/test/java/uk/gov/companieshouse/orders/api/service/CheckoutServiceTest.java [hunk 2]
```diff
@@ -57,12 +67,43 @@ void createCheckoutPopulatesAndSavesItem() {
...
+        when(linksGeneratorService.generateCheckoutLinks(any(String.class))).thenReturn(checkoutLinks);
+
+        serviceUnderTest.createCheckout(new Certificate(), ERIC_IDENTITY_VALUE, ERIC_AUTHORISED_USER_VALUE);
+        verify(checkoutRepository).save(argCaptor.capture());
+
+        verify(etagGeneratorService, times(1)).generateEtag();
+        verify(linksGeneratorService, times(1)).generateCheckoutLinks(any(String.class));
+        assertEquals(LINKS_SELF, argCaptor.getValue().getData().getLinks().getSelf());
+        assertEquals(LINKS_PAYMENT, argCaptor.getValue().getData().getLinks().getPayment());
     }
```
Comment (VaughanJackson123) [inline src/test/java/uk/gov/companieshouse/orders/api/service/CheckoutServiceTest.java]: Can we make an assertion about the value of the etag here too?

Comment (nabeelamirch) [reply src/test/java/uk/gov/companieshouse/orders/api/service/CheckoutServiceTest.java]: Oh yes, I missed that. I will add that in.

File: src/main/java/uk/gov/companieshouse/orders/api/model/OrderData.java (hunk 1 above)
Comment (VaughanJackson123) [reply src/main/java/uk/gov/companieshouse/orders/api/model/OrderData.java]: Ah OK, clear now! 👍 .

**Pull request merged**